tests/tests.rs

```diff
@@ -0,0 +1,109 @@
+extern crate froggy;
+
+use froggy::Storage;
+
+#[test]
+fn change_by_pointer() {
+    let storage = Storage::new();
+    {
+        let mut s = storage.write();
+        for &v in [5 as i32, 7, 4, 6, 7].iter() {
+            s.create(v);
+        }
+    }
+    assert_eq!(storage.read().iter().count(), 5);
+    assert_eq!(*storage.read().iter().nth(0).unwrap(), 5);
+    assert_eq!(*storage.read().iter().nth(1).unwrap(), 7);
+    let ptr = {
+        let r = storage.read();
+        let item = r.iter().find(|v| **v == 4).unwrap();
+        r.pin(&item)
+    };
+    let mut w = storage.write();
+    w[&ptr] = 350 as i32;
+    assert_eq!(w[&ptr], 350);
+}
+
+#[test]
+#[should_panic]
+#[allow(unused_variables)]
+fn write_and_read() {
+    let storage: Storage<i32> = Storage::new();
+    let write_lock = storage.write();
+    let read_lock = storage.read();
+}
+
+#[test]
+fn iterators() {
+    let storage = Storage::new();
+    {
+        let mut w = storage.write();
+        for i in 0..5 {
+            w.create(i * 3 as i32);
+        }
+    }
+    assert_eq!(storage.write().iter().count(), 5);
+    assert_eq!(storage.write().iter_alive().count(), 0);
+    let ptr = {
+        let r = storage.read();
+        let item = r.iter().nth(0).unwrap();
+        r.pin(&item)
+    };
+    assert_eq!(storage.read()[&ptr], 0);
+    assert_eq!(storage.write().iter().count(), 5);
+    assert_eq!(storage.write().iter_alive().count(), 1);
+}
+
+#[test]
+fn pointer_iter() {
+    let storage = Storage::new();
+    {
+        let mut w = storage.write();
+        for i in 0..5 {
+            w.create(i as i32);
```

Comment:
Are you talking about filling storage this way? Or about iterating over Pointers?